File: android/src/main/java/org/ligi/passandroid/printing/PrintHelper.java
package org.ligi.passandroid.printing;

import android.annotation.TargetApi;
import android.content.Context;
import android.os.Build;
import android.print.PrintManager;
import org.ligi.passandroid.R;
import org.ligi.passandroid.model.pass.Pass;

public class PrintHelper {

    @TargetApi(Build.VERSION_CODES.KITKAT)
    public static void doPrint(Context context, Pass pass) {
        final PrintManager printManager = (PrintManager) context.getSystemService(Context.PRINT_SERVICE);
        final String jobName = context.getString(R.string.app_name) + " print of " + pass.getDescription();
        printManager.print(jobName, new PassPrintDocumentAdapter(context, pass, jobName), null);
    }

}

File: android/src/main/java/org/ligi/passandroid/printing/PrintHelper.kt
package org.ligi.passandroid.printing

import android.annotation.TargetApi
import android.content.Context
import android.os.Build
import android.print.PrintManager
import org.ligi.passandroid.R
import org.ligi.passandroid.model.pass.Pass

@TargetApi(Build.VERSION_CODES.KITKAT)
fun doPrint(context: Context, pass: Pass) {
    val printManager = context.getSystemService(Context.PRINT_SERVICE) as PrintManager
    val jobName = context.getString(R.string.app_name) + " print of " + pass.description
    printManager.print(jobName, PassPrintDocumentAdapter(context, pass, jobName), null)
}
